src/parser/InlineParser.ts

```diff
@@ -24,28 +24,17 @@ export class InlineParser<InlineGroupData, InlineData, ParentData = undefined> {
 	private current?: InlineGroupData;
 
 	// Map of sugar start character to sugar definition:
-	private sugars?: Map<string, NamedSugar>;
+	private sugars?: Map<string, Sugar>;
 	private isRaw?: boolean;
 
 	constructor(private readonly handler: InlineHandler<InlineGroupData, InlineData, ParentData>) {
-		const sugars = handler.getAllSugars();
-		this.regex = new RegExp(
-			"(" +
-				[
-					/#[^ \[]+\[?/.source,
-					/\\./.source,
-					...[this.inlineTag.separator ? [escapeRegExp(this.inlineTag.separator)] : []],
-					...[
-						this.inlineTag.end,
-						...sugars.map(_ => _.start),
-						...sugars.map(_ => _.separator).filter((_): _ is string => Boolean(_)),
-						...sugars.map(_ => _.end)
-					].map(escapeRegExp)
-				]
-					.map(_ => `(?:${_})`)
-					.join("|") +
-				")",
-			"g"
+		const customTokens = handler.allSugars.map(_ => _.syntax).flatMap(Object.values);
+		this.regex = regexpUnion(
+			/\\./,
+			/#[^ \[]+\[?/,
+			stringToRegexp(this.inlineTagSyntax.separator),
+			stringToRegexp(this.inlineTagSyntax.end),
+			...unique(customTokens.sort((a, b) => a.length - b.length)).map(stringToRegexp)
```

Comment:
The implementation of `unique` converts to `Set<T>`, and then to `T[]`. This conversion is not guaranteed to preserve ordering, as a `Set<T>` has no notion of ordering.

This line should be:

```typescript
...unique(customTokens).sort((a, b) => a.length - b.length).map(stringToRegexp)
```

Reply:
Apparently this isn't true, [`Set` preserves insertion order!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set). TIL. Approving this PR.